### PR TITLE
fix non-dummy objectives not updating

### LINKED ISSUE #5370: Health objective in a non-default scoreboard isn't updated
Here's some code for context.
```java
@EventHandler
public void onJoin(PlayerJoinEvent e) {
    e.getPlayer().setScoreboard(Bukkit.getScoreboardManager().getNewScoreboard());
    Objective healthObjective = e.getPlayer().getScoreboard().getObjective(DisplaySlot.BELOW_NAME);
    if (healthObjective == null) {
        healthObjective = e.getPlayer().getScoreboard().registerNewObjective("health", "health", Component.text("HP"));
        healthObjective.setDisplaySlot(DisplaySlot.BELOW_NAME);
    }
}
```
When a player joins, it creates a new scoreboard and applies it to them.
Then, it creates a new objective that should show players' health below their name.

### What behaviour is expected:
Players' health should show below their name and update when damage is taken or health is regenerated.

### What behaviour is observed:
It shows 0 health on login and never updates.

### Steps/models to reproduce:
Just create a plugin with that snippet and add it to your Paper installation.

### Plugin list:
Only my test plugin.

### Paper version:
`This server is running Paper version git-Paper-551 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)`

### Anything else:
This bug only happens since Paper went on an Adventure in 4e958e229f9f448a10dbc610bab460dbf222902e.
I tested with builds 472, 473 (Adventure) and 551.
It works fine in 472, but the issue is present in both 473 and 551 (latest at the time of writing).

Using a custom scoreboard created with getNewScoreboard() is important. It works fine using the default scoreboard.

In the above code I set a display name for the objective using Adventure's Component but this doesn't matter. Using Component.text, Component.empty, null or using deprecated String names do not affect the outcome.

### PR DESCRIPTION
**Pull request by Machine-Maker**
fixes #5370

### CODE REVIEW

File: Spigot-Server-Patches/0574-Lazily-track-plugin-scoreboards-by-default.patch [hunk 1]
```diff
@@ -39,7 +39,20 @@ index 6ea491f6308317059c4bc6735abbdce370df0f34..88bc3fbfa3bfb27cc4f9003a99cbce6e
  
      CraftScoreboard(Scoreboard board) {
          this.board = board;
-@@ -66,6 +67,12 @@ public final class CraftScoreboard implements org.bukkit.scoreboard.Scoreboard {
+@@ -45,6 +46,12 @@ public final class CraftScoreboard implements org.bukkit.scoreboard.Scoreboard {
+         Validate.isTrue(name.length() <= 16, "The name '" + name + "' is longer than the limit of 16 characters");
+         Validate.isTrue(board.getObjective(name) == null, "An objective of name '" + name + "' already exists");
+         CraftCriteria craftCriteria = CraftCriteria.getFromBukkit(criteria);
++        // Paper start
```
Comment (Proximyst) [inline Spigot-Server-Patches/0574-Lazily-track-plugin-scoreboards-by-default.patch]: Add why this is added here

**Review by Proximyst — DISMISSED** [review]

File: Spigot-Server-Patches/0574-Lazily-track-plugin-scoreboards-by-default.patch (hunk 1 above)
Comment (Machine-Maker) [reply Spigot-Server-Patches/0574-Lazily-track-plugin-scoreboards-by-default.patch]: Done

Comment (Proximyst) [reply Spigot-Server-Patches/0574-Lazily-track-plugin-scoreboards-by-default.patch]: I more meant an extra comment pointing out how the block comment you showed internally broke this 😅

**Review by kashike — APPROVED** [review]

File: Spigot-Server-Patches/0574-Lazily-track-plugin-scoreboards-by-default.patch (hunk 1 above)
Comment (Machine-Maker) [reply Spigot-Server-Patches/0574-Lazily-track-plugin-scoreboards-by-default.patch]: Ok, should be there now

**Review by kennytv — APPROVED** [review]

**Pull request merged**